content/en/docs/setup/install/operator/test.sh

```diff
@@ -13,30 +15,70 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+
+# @setup profile=none
+
 set -e
 set -u
-
 set -o pipefail
 
-# @setup profile=none
+function testOperatorDeployWatchNs(){
+    # print out body of the function and execute with flag
+    # this is to avoid using the default public registry
+    $(type snip_deploy_istio_operator_watch_ns | sed '1,3d;$d') --hub "$HUB"
+    _wait_for_deployment istio-operator istio-operator
 
-snip_create_istio_operator
-_wait_for_deployment istio-operator istio-operator
+    # cleanup required for next steps
+    istioctl uninstall -y --purge
+    kubectl delete ns istio-operator istio-namespace1 istio-namespace2
+}
 
-snip_create_demo_profile
-sleep 30s
-_wait_for_deployment istio-system istiod
+function testOperatorDeployHelm(){
+    snip_create_ns_istio_operator
+    snip_deploy_istio_operator_helm
+    _wait_for_deployment istio-operator istio-operator
 
-# shellcheck disable=SC2154
-_verify_like snip_kubectl_get_svc "$snip_kubectl_get_svc_out"
+    # cleanup required for next steps
+    helm uninstall istio-operator -n istio-operator
+    kubectl delete ns istio-operator
+}
 
-# shellcheck disable=SC2154
-_verify_like snip_kubectl_get_pods "$snip_kubectl_get_pods_out"
+function testOperatorDeploy(){
+    $(type snip_deploy_istio_operator | sed '1,3d;$d') --hub "$HUB"
+    _wait_for_deployment istio-operator istio-operator
+}
 
-snip_update_operator
-sleep 30s
-_verify_contains snip_kubectl_get_svc "egressgateway"
+function testInstallIstioDemo(){
+    snip_install_istio_demo_profile
+    sleep 30s
+    _wait_for_deployment istio-system istiod
+    _verify_like snip_kubectl_get_svc "$snip_kubectl_get_svc_out"
+    _verify_like snip_kubectl_get_pods "$snip_kubectl_get_pods_out"
+}
+
+function testUpdateProfileDefaultEgress(){
+    snip_update_to_default_profile_egress
+    sleep 30s
+    _verify_contains snip_kubectl_get_svc "egressgateway"
+}
+
+testOperatorDeployWatchNs
+
+testOperatorDeployHelm
+
+testOperatorDeploy
+
+testInstallIstioDemo
+
+snip_update_to_default_profile
+
+testUpdateProfileDefaultEgress
+
+_verify_like snip_verify_operator_cr "$snip_verify_operator_cr_out"
 
 # @cleanup
 snip_cleanup
-kubectl delete ns istio-operator --grace-period=0 --force
+
+# Everything should be removed once cleanup completes. Use a small
+# timeout for comparing cluster snapshots before/after the test.
+export VERIFY_TIMEOUT=20
```

Comment:
I don't think there's any advantage to decreasing the verify timeout, but it risks causing the test to become flaky, i.e., fail when the build machine is overly loaded. The timeout only affects the failure case after repeatedly trying every 1s, so normally it would complete much sooner than the default 2 mins.

Reply:
Thanks, I will remove this.

Reply:
this was addressed in 6941f93a050425a64dfd08cdd45cf8452587b22c